relay/sources/relayd/src/lib.rs

```diff
@@ -82,13 +82,19 @@ use tracing_subscriber::{
     reload::Handle,
 };
 
-/// Sets exit code based on error type
-/// None means unspecified error or panic
-pub fn return_code(e: Option<&Error>) -> i32 {
-    match e {
-        Some(Error::ConfigurationParsing(_)) => 2,
-        Some(_) | None => 1,
-    }
+/// Exit with code based on error type
+///
+/// * `Ok(())` means expected shutdown
+/// * `Error(None)` means unexpected crash or panic
+/// * `Error(Some(e))` means shutdown with given error
+pub fn exit_with(r: Result<(), Option<&Error>>) -> ! {
+    exit(match r {
+        Ok(()) => 0,
+        Err(None) => 1,
+        Err(Some(Error::ServerHaltedUnexpectedly)) => 1,
+        Err(Some(Error::ConfigurationParsing(_))) => 2,
+        Err(Some(_)) => 3,
```

Comment:
What are those other errors and why are they considered to be unfixable by a restart ?

Reply:
Ill update the PR with more details about errors.